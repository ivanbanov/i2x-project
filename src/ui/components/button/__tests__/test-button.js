import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../';

jest.mock('../styles.styl', () => ({}));

describe('Button', () => {
  it('should render properly', () => {
    const component = renderer.create(
      <Button
        className="button"
        value="Button"
        disabled
        loading
        block
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const fn = jest.fn();

    const component = renderer.create(
      <Button
        onClick={fn}
        value="onClick"
      />
    );

    const tree = component.toJSON();
    tree.props.onClick();

    expect(fn).toHaveBeenCalled();
  });

  it('should render as link element <a>', () => {
    const component = renderer.create(
      <Button
        href="http://foobar.com"
        value="link"
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
