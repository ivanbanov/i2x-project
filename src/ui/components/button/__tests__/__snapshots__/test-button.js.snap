// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as link element <a> 1`] = `
<a
  className=""
  href="http://foobar.com"
  onClick={[Function]}
>
  link
</a>
`;

exports[`Button should render properly 1`] = `
<button
  className="undefined undefined undefined"
  onClick={[Function]}
>
  Button
</button>
`;
